scripts/rust-linter.sh

```diff
@@ -9,4 +9,5 @@ $CARGO clippy --all --all-targets -- -D warnings \
     -A clippy::option_env_unwrap \
     -A clippy::redundant-guards \
     -A clippy::suspicious-doc-comments \
+    -A clippy::useless-format \
```

Comment:
why do we need this?
Btw seems like we have a lot of whitelists here, any reason for this?

Reply:
As we update rustc/clippy versions, new lints come in. Some of them are not easily fixable, and for some we just don't have enough time.

As for "useless-format", it is a newer one and a quite annoying.

E.g. you have a list like:
let args = [
    format!("--abc={a}"),
    format!("--def"),
    format!("--qwe={b}"),
];

It is nicely and uniformly formatted, and does not cause any troubles. Now, clippy started to complain about it, suggesting to replace format!("--def") with "--def".to_string(). I don't see much sense in this.